### ISSUE TITLE
Warn on unknown language extensions

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#1093]
See https://github.com/ndmitchell/hlint/issues/223#issuecomment-665951432 - if you type:

```
hlint -XQuasiQuoe Sample.hs
```

It doesn't warn or anything. It probably should at least warn. Not necessarily as a hint warn, but just `putStrLn $ "Warning: Unknown extension, QuasiQuoe"`.

**Comment by shayne-fletcher** [comment]
Ouch. For when we get to it, I think I see where we need to intervene: function `getExtensions`, `CmdLine.hs:333`.

**Comment by shayne-fletcher** [comment]
> Ouch. For when we get to it, I think I see where we need to intervene: function `getExtensions`, `CmdLine.hs:333`.

If that is indeed the right place, maybe `error` here?
  - It'll be easier to implement;
  - Consistent with the expected behavior of an unrecognized argument.

**Comment by ndmitchell** [comment]
My first inclination was not to error, since if GHC adds a new extension we don't know about then we'd fail. But honestly, now we track GHC so closely, I'm not sure it's a real concern.

**Comment by shayne-fletcher** [comment]
> My first inclination was not to error, since if GHC adds a new extension we don't know about then we'd fail. But honestly, now we track GHC so closely, I'm not sure it's a real concern.

I agree with this; don't think that's a concern.

**Comment by shayne-fletcher** [comment]
> My first inclination was not to error, since if GHC adds a new extension we don't know about then we'd fail. 

On reflection, I think that's the behavior we want if we find ourselves in this case too since it's a very much a pragma we can't satisfy!

**Comment by shayne-fletcher** [comment]
> > My first inclination was not to error, since if GHC adds a new extension we don't know about then we'd fail.
> 
> On reflection, I think that's the behavior we want if we find ourselves in this case too since it's a very much a pragma we can't satisfy!

Ah but I see your point that warning would be preferable in this case - you'd want to say "I can't do that" and proceed to analyze what you can. Ok, scratch that and in any case as noted above, we don't intend to ever be in this path anyway 😉
